src/lib/Microsoft.Health.Events/EventConsumers/Service/EventBatchingService.cs

```diff
@@ -64,7 +68,7 @@ public bool EventPartitionExists(string partitionId)
 
         private EventPartition CreatePartitionIfMissing(string partitionId, DateTime initTime, TimeSpan flushTimespan)
         {
-            return _eventPartitions.GetOrAdd(partitionId, new EventPartition(partitionId, initTime, flushTimespan, _logger));
+            return _eventPartitions.GetOrAdd(partitionId, new EventPartition(partitionId, initTime, flushTimespan, _maxEvents, _logger));
```

Comment:
I am not sure of the logic flow.  After reviewing the code it appears that partitions are initially configured with the current default batch size.  Then the batch size will be adjusted on the first flush of that partition?  While this can work I think is fairly complicated which may lead to edge cases/bugs we aren't considering.

One potential issue I see is we will have multiple threads potentially reading/adjusting the max partition value.  This means we should interlocks at min on the partition count value the partition classes are using.

To simplify things I would propose the following.  Create a new class like PartitionSync (or a better :) ) so we can pass by ref to the partitions.  This class would contain the current max events to read.  We could add methods to pull the current value in a thread safe manner and used in the ConsumeEvent and FlushMax events.  The main advantage is you don't need to update the partition value, you can just reference the value the EventBatchingService maintains in a thread safe manner.

Reply:
Updating to just have the max events in the batching class, rather than the partition class, and implemented some locking.